### PR TITLE
OpsGenie notification backend

### PR DESCRIPTION
**Pull request by rgbj**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
It adds a new notification backend alloiwng to send notifications to OpsGenie.

* **What is the current behavior?** (You can also link to an open issue here)
Notifications can be routed to slack, TaT, or a generic webhook.

* **What is the new behavior (if this is a feature change)?**
Notifications can be routed to slack, TaT, a generic webhook, or OpsGenie.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Other information**:

### CODE REVIEW

File: pkg/notify/opsgenie/opsgenie.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// NewOpsGenieNotificationSender instantiates a NotificationSender
+func NewOpsGenieNotificationSender(zone, apikey string) (*NotificationSender, error) {
+	zonesToApiUrls := map[string]client.ApiUrl{
+		ZoneDefault: client.API_URL,
+		ZoneEU:      client.API_URL_EU,
+		ZoneSandbox: client.API_URL_SANDBOX,
+	}
+	apiUrl, present := zonesToApiUrls[zone]
+	if !present {
+		return nil, errors.NotFoundf("opsgenie zone %q not found", zone)
```
Comment (rbeuque74) [inline pkg/notify/opsgenie/opsgenie.go:38]:
```suggestion
		return nil, errors.NotFoundf("opsgenie zone %q", zone)
```

juju/errors already suffix with `not found`

**Review by rbeuque74 — APPROVED** [review]

**Review by rbeuque74 — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report test#1130.0 ✘  
* tests and lint
  * golangci-lint ✘  
  * unit-test and integration-tests ✔

**Review by rbeuque74 — CHANGES_REQUESTED** [review]
> CDS Report test#1130.0 ✘
> 
>     * tests and lint
>       
>       * golangci-lint ✘
>       * unit-test and integration-tests heavy_check_mark

Golang-Ci lint failed to pass
```
+ golangci-lint run
utask.go:151:2: struct field `ApiKey` should be `APIKey` (golint)
	ApiKey string `json:"api_key"`
	^
pkg/notify/opsgenie/opsgenie.go:31:2: var `zonesToApiUrls` should be `zonesToAPIUrls` (golint)
	zonesToApiUrls := map[string]client.ApiUrl{
	^
pkg/notify/opsgenie/opsgenie.go:36:2: var `apiUrl` should be `apiURL` (golint)
	apiUrl, present := zonesToApiUrls[zone]
	^
pkg/notify/opsgenie/opsgenie.go:62:29: SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use (staticcheck)
	_, err := ns.client.Create(nil, req)
	                           ^
```

**Comment by ovh-cds** [comment]
CDS Report test#1132.0 ✘  
* tests and lint
  * golangci-lint ✘  
  * unit-test and integration-tests ✔

**Review by rbeuque74 — CHANGES_REQUESTED** [review]
```
pkg/notify/opsgenie/opsgenie.go:4: File is not `goimports`-ed (goimports)
	"context"
````

I would suggest you to run `golangci-lint run` on your development machine, so you can expect where the CI will fail.

File: pkg/notify/opsgenie/opsgenie.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+// Send dispatches a notify.Message to OpsGenie
+func (ns *NotificationSender) Send(m *notify.Message, name string) {
+	req := &alert.CreateAlertRequest{
+		Message:     m.MainMessage,
+		Description: m.MainMessage,
+		Details:     m.Fields,
+	}
+
+	ctx := context.Background()
+	_, err := ns.client.Create(ctx, req)
```
Comment (Alkorin) [inline pkg/notify/opsgenie/opsgenie.go:63]: Missing timeout

Comment (rgbj) [reply pkg/notify/opsgenie/opsgenie.go:67]: > I would suggest you to run `golangci-lint run` on your development machine, so you can expect where the CI will fail.

will do, now that I found the right tool! :) Sorry for the noise

**Review by rbeuque74 — APPROVED** [review]

**Pull request merged**